src/simplnx/Utilities/Parsing/HDF5/Readers/DatasetReader.hpp

```diff
@@ -113,12 +113,12 @@ class SIMPLNX_EXPORT DatasetReader : public ObjectReader
 
   /**
    * @brief Reads the dataset into the given span. Requires the span to be the
-   * correct size. Returns false if unable to read.
+   * correct size. Returns invalid result if unable to read.
    * @tparam T
    * @param data
    */
   template <class T>
-  bool readIntoSpan(nonstd::span<T> data) const;
+  Result<> readIntoSpan(nonstd::span<T> data, const std::optional<std::vector<hsize_t>>& start = std::nullopt, const std::optional<std::vector<hsize_t>>& count = std::nullopt) const;
```

Comment:
The documentation should be updated for the new arguments. It's unclear what they're used for.